Error occurred when inserting time value with microseconds into ShardingSphere-Proxy PostgreSQL

## Bug Report

### Which version of ShardingSphere did you use?

master - 70c02a332868c971e34a326995bb8f9a100d5f35

This may affect all versions <= 5.3.1.

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

ShardingSphere-Proxy PostgreSQL

### Expected behavior

Insert succeeded.

### Actual behavior

Client log:
```
Exception in thread "main" org.postgresql.util.PSQLException: ERROR: Error at index 2 in: "19.65948"
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2675)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2365)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:355)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:490)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:408)
	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:166)
	at org.postgresql.jdbc.PgPreparedStatement.executeUpdate(PgPreparedStatement.java:134)
```

Proxy log:
```
[ERROR] 2023-01-28 15:40:32.955 [Connection-2-ThreadExecutor] o.a.s.p.f.c.CommandExecutorTask - Exception occur: 
java.lang.NumberFormatException: Error at index 2 in: "32.949965"
	at java.base/java.lang.NumberFormatException.forCharSequence(NumberFormatException.java:83)
	at java.base/java.lang.Integer.parseInt(Integer.java:751)
	at java.sql/java.sql.Time.valueOf(Time.java:107)
	at org.apache.shardingsphere.db.protocol.postgresql.packet.command.query.extended.bind.PostgreSQLComBindPacket.getTextParameters(PostgreSQLComBindPacket.java:130)
	at org.apache.shardingsphere.db.protocol.postgresql.packet.command.query.extended.bind.PostgreSQLComBindPacket.getTextParameters(PostgreSQLComBindPacket.java:99)
	at org.apache.shardingsphere.db.protocol.postgresql.packet.command.query.extended.bind.PostgreSQLComBindPacket.readParameters(PostgreSQLComBindPacket.java:80)
	at org.apache.shardingsphere.proxy.frontend.postgresql.command.query.extended.bind.PostgreSQLComBindExecutor.execute(PostgreSQLComBindExecutor.java:51)
	at org.apache.shardingsphere.proxy.frontend.postgresql.command.query.extended.PostgreSQLAggregatedCommandExecutor.execute(PostgreSQLAggregatedCommandExecutor.java:41)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:110)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:77)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

### Example codes for reproduce this issue (such as a github link).

```java
try (Statement statement = connection.createStatement()) {
    statement.executeUpdate("drop table if exists datetime_pg");
    statement.executeUpdate("create table datetime_pg (val_time time not null)");
}
try (PreparedStatement preparedStatement = connection.prepareStatement("insert into datetime_pg values (?)")) {
    preparedStatement.setObject(1, LocalTime.now());
    preparedStatement.executeUpdate();
}
```

--- a/db-protocol/opengauss/src/main/java/org/apache/shardingsphere/db/protocol/opengauss/packet/command/query/extended/bind/OpenGaussComBatchBindPacket.java
+++ b/db-protocol/opengauss/src/main/java/org/apache/shardingsphere/db/protocol/opengauss/packet/command/query/extended/bind/OpenGaussComBatchBindPacket.java
@@ -26,12 +26,12 @@ import org.apache.shardingsphere.db.protocol.postgresql.constant.PostgreSQLValue
 import org.apache.shardingsphere.db.protocol.postgresql.packet.command.query.extended.PostgreSQLColumnType;
 import org.apache.shardingsphere.db.protocol.postgresql.packet.command.query.extended.bind.protocol.PostgreSQLBinaryProtocolValue;
 import org.apache.shardingsphere.db.protocol.postgresql.packet.command.query.extended.bind.protocol.PostgreSQLBinaryProtocolValueFactory;
+import org.apache.shardingsphere.db.protocol.postgresql.packet.command.query.extended.bind.protocol.PostgreSQLTextTimeUtils;
 import org.apache.shardingsphere.db.protocol.postgresql.packet.command.query.extended.bind.protocol.PostgreSQLTextTimestampUtils;
 import org.apache.shardingsphere.db.protocol.postgresql.payload.PostgreSQLPacketPayload;
 
 import java.math.BigDecimal;
 import java.sql.Date;
-import java.sql.Time;
 import java.util.ArrayList;
 import java.util.List;
 
@@ -139,7 +139,7 @@ public final class OpenGaussComBatchBindPacket extends OpenGaussCommandPacket {
             case POSTGRESQL_TYPE_DATE:
                 return Date.valueOf(textValue);
             case POSTGRESQL_TYPE_TIME:
-                return Time.valueOf(textValue);
+                return PostgreSQLTextTimeUtils.parse(textValue);
             case POSTGRESQL_TYPE_TIMESTAMP:
             case POSTGRESQL_TYPE_TIMESTAMPTZ:
                 return PostgreSQLTextTimestampUtils.parse(textValue);

--- a/db-protocol/postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/PostgreSQLComBindPacket.java
+++ b/db-protocol/postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/PostgreSQLComBindPacket.java
@@ -25,13 +25,13 @@ import org.apache.shardingsphere.db.protocol.postgresql.packet.command.PostgreSQ
 import org.apache.shardingsphere.db.protocol.postgresql.packet.command.query.extended.PostgreSQLColumnType;
 import org.apache.shardingsphere.db.protocol.postgresql.packet.command.query.extended.bind.protocol.PostgreSQLBinaryProtocolValue;
 import org.apache.shardingsphere.db.protocol.postgresql.packet.command.query.extended.bind.protocol.PostgreSQLBinaryProtocolValueFactory;
+import org.apache.shardingsphere.db.protocol.postgresql.packet.command.query.extended.bind.protocol.PostgreSQLTextTimeUtils;
 import org.apache.shardingsphere.db.protocol.postgresql.packet.command.query.extended.bind.protocol.PostgreSQLTextTimestampUtils;
 import org.apache.shardingsphere.db.protocol.postgresql.packet.identifier.PostgreSQLIdentifierTag;
 import org.apache.shardingsphere.db.protocol.postgresql.payload.PostgreSQLPacketPayload;
 
 import java.math.BigDecimal;
 import java.sql.Date;
-import java.sql.Time;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
@@ -127,7 +127,7 @@ public final class PostgreSQLComBindPacket extends PostgreSQLCommandPacket {
             case POSTGRESQL_TYPE_DATE:
                 return Date.valueOf(textValue);
             case POSTGRESQL_TYPE_TIME:
-                return Time.valueOf(textValue);
+                return PostgreSQLTextTimeUtils.parse(textValue);
             case POSTGRESQL_TYPE_TIMESTAMP:
             case POSTGRESQL_TYPE_TIMESTAMPTZ:
                 return PostgreSQLTextTimestampUtils.parse(textValue);

--- a/db-protocol/postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLTextTimeUtils.java
+++ b/db-protocol/postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLTextTimeUtils.java
@@ -1,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.db.protocol.postgresql.packet.command.query.extended.bind.protocol;
+
+import org.apache.shardingsphere.infra.util.exception.external.sql.type.generic.UnsupportedSQLOperationException;
+
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Utils for PostgreSQL time.
+ */
+public final class PostgreSQLTextTimeUtils {
+    
+    private static final DateTimeFormatter POSTGRESQL_TIME_FORMATTER = DateTimeFormatter.ofPattern(
+            "[HH:mm:ss][HHmmss][HH:mm][HHmm]"
+                    + "[.SSSSSSSSS][.SSSSSSSS][.SSSSSSS][.SSSSSS][.SSSSS][.SSSS][.SSS][.SS][.S]"
+                    + "[ ]"
+                    + "[XXXXX][XXXX][XXX][XX][X]");
+    
+    /**
+     * Parse time in PostgreSQL text format.<br>
+     * NOTICE: PostgreSQL allows 6 fractional digits retained in the seconds field, which is unsupported by {@link java.sql.Time}.
+     *
+     * @param value text value to be parsed
+     * @return time
+     */
+    public static LocalTime parse(final String value) {
+        try {
+            return POSTGRESQL_TIME_FORMATTER.parse(value, LocalTime::from);
+        } catch (final DateTimeParseException ignored) {
+            throw new UnsupportedSQLOperationException("Unsupported time format: [" + value + "]");
+        }
+    }
+}

--- a/db-protocol/postgresql/src/test/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLTextTimeUtilsTest.java
+++ b/db-protocol/postgresql/src/test/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLTextTimeUtilsTest.java
@@ -1,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.db.protocol.postgresql.packet.command.query.extended.bind.protocol;
+
+import lombok.RequiredArgsConstructor;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+import java.time.LocalTime;
+import java.util.Arrays;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@RunWith(Parameterized.class)
+@RequiredArgsConstructor
+public final class PostgreSQLTextTimeUtilsTest {
+    
+    private final String input;
+    
+    private final LocalTime expected;
+    
+    @Parameters(name = "{0}")
+    public static Iterable<Object[]> textValues() {
+        return Arrays.asList(
+                new Object[]{"2323", LocalTime.of(23, 23, 0)},
+                new Object[]{"23:23", LocalTime.of(23, 23, 0)},
+                new Object[]{"232323", LocalTime.of(23, 23, 23)},
+                new Object[]{"23:23:23", LocalTime.of(23, 23, 23)},
+                new Object[]{"23:23:23.1", LocalTime.of(23, 23, 23, 100_000_000)},
+                new Object[]{"23:23:23.12", LocalTime.of(23, 23, 23, 120_000_000)},
+                new Object[]{"23:23:23.123", LocalTime.of(23, 23, 23, 123_000_000)},
+                new Object[]{"23:23:23.123+08", LocalTime.of(23, 23, 23, 123_000_000)},
+                new Object[]{"23:23:23.123+08", LocalTime.of(23, 23, 23, 123_000_000)},
+                new Object[]{"23:23:23.12345", LocalTime.of(23, 23, 23, 123_450_000)},
+                new Object[]{"23:23:23.12345+0800", LocalTime.of(23, 23, 23, 123_450_000)},
+                new Object[]{"23:23:23.12345+0800", LocalTime.of(23, 23, 23, 123_450_000)},
+                new Object[]{"23:23:23.12345+08:00:00", LocalTime.of(23, 23, 23, 123_450_000)},
+                new Object[]{"23:23:23.12345+08:00:00", LocalTime.of(23, 23, 23, 123_450_000)},
+                new Object[]{"23:23:23.12345+08:00:00", LocalTime.of(23, 23, 23, 123_450_000)},
+                new Object[]{"23:23:23.123456", LocalTime.of(23, 23, 23, 123_456_000)},
+                new Object[]{"23:23:23.1234567", LocalTime.of(23, 23, 23, 123_456_700)},
+                new Object[]{"23:23:23.12345678", LocalTime.of(23, 23, 23, 123_456_780)},
+                new Object[]{"23:23:23.123456789", LocalTime.of(23, 23, 23, 123_456_789)},
+                new Object[]{"23:23:23.123456 +08:00", LocalTime.of(23, 23, 23, 123_456_000)},
+                new Object[]{"23:23:23.123456 -08:00", LocalTime.of(23, 23, 23, 123_456_000)},
+                new Object[]{"23:23:23.123456789 +08:00", LocalTime.of(23, 23, 23, 123_456_789)},
+                new Object[]{"23:23:23.123456", LocalTime.of(23, 23, 23, 123_456_000)});
+    }
+    
+    @Test
+    public void assertParse() {
+        assertThat(PostgreSQLTextTimeUtils.parse(input), is(expected));
+    }
+}

